internal/cephes/incbeta.go

```diff
@@ -0,0 +1,311 @@
+	if y < minLog {
+		t = 0.0
+	} else {
+		t = math.Exp(y)
+	}
+
+	return transformT(t, flag)
+}
+
+func transformT(t float64, flag int) float64 {
+	if flag == 1 {
+		if t <= machEp {
+			t = 1.0 - machEp
+		} else {
+			t = 1.0 - t
+		}
+	}
+	return t
+}
+
+// incbcf is the continued fraction expansion #1 for incomplete beta integral
+func incbcf(a, b, x float64) float64 {
+	var xk, pk, pkm1, pkm2, qk, qkm1, qkm2 float64
+	var k1, k2, k3, k4, k5, k6, k7, k8 float64
+	var r, t, ans, thresh float64
+	var n int
+
+	k1 = a
+	k2 = a + b
+	k3 = a
+	k4 = a + 1.0
+	k5 = 1.0
+	k6 = b - 1.0
+	k7 = k4
+	k8 = a + 2.0
+
+	pkm2 = 0.0
+	qkm2 = 1.0
+	pkm1 = 1.0
+	qkm1 = 1.0
+	ans = 1.0
+	r = 1.0
+	thresh = 3.0 * machEp
+
+	for n = 0; n <= 300; n++ {
+
+		xk = -(x * k1 * k2) / (k3 * k4)
+		pk = pkm1 + pkm2*xk
+		qk = qkm1 + qkm2*xk
+		pkm2 = pkm1
+		pkm1 = pk
+		qkm2 = qkm1
+		qkm1 = qk
+
+		xk = (x * k5 * k6) / (k7 * k8)
+		pk = pkm1 + pkm2*xk
+		qk = qkm1 + qkm2*xk
+		pkm2 = pkm1
+		pkm1 = pk
+		qkm2 = qkm1
+		qkm1 = qk
+
+		if qk != 0 {
+			r = pk / qk
+		}
+		if r != 0 {
+			t = math.Abs((ans - r) / r)
+			ans = r
+		} else {
+			t = 1.0
+		}
+
+		if t < thresh {
+			return ans
+		}
+
+		k1 += 1.0
+		k2 += 1.0
+		k3 += 2.0
+		k4 += 2.0
+		k5 += 1.0
+		k6 -= 1.0
+		k7 += 2.0
+		k8 += 2.0
+
+		if (math.Abs(qk) + math.Abs(pk)) > big {
+			pkm2 *= biginv
+			pkm1 *= biginv
+			qkm2 *= biginv
+			qkm1 *= biginv
+		}
+		if (math.Abs(qk) < biginv) || (math.Abs(pk) < biginv) {
+			pkm2 *= big
+			pkm1 *= big
+			qkm2 *= big
+			qkm1 *= big
+		}
+	}
+
+	return ans
+}
+
+// incbd is the continued fraction expansion #2 for incomplete beta integral
+func incbd(a, b, x float64) float64 {
+	var xk, pk, pkm1, pkm2, qk, qkm1, qkm2 float64
+	var k1, k2, k3, k4, k5, k6, k7, k8 float64
+	var r, t, ans, z, thresh float64
+	var n int
+
+	k1 = a
+	k2 = b - 1.0
+	k3 = a
+	k4 = a + 1.0
+	k5 = 1.0
+	k6 = a + b
+	k7 = a + 1.0
+	k8 = a + 2.0
+
+	pkm2 = 0.0
+	qkm2 = 1.0
+	pkm1 = 1.0
+	qkm1 = 1.0
+	z = x / (1.0 - x)
+	ans = 1.0
+	r = 1.0
+	thresh = 3.0 * machEp
+	for n = 0; n <= 300; n++ {
+
+		xk = -(z * k1 * k2) / (k3 * k4)
+		pk = pkm1 + pkm2*xk
+		qk = qkm1 + qkm2*xk
+		pkm2 = pkm1
+		pkm1 = pk
+		qkm2 = qkm1
+		qkm1 = qk
+
+		xk = (z * k5 * k6) / (k7 * k8)
+		pk = pkm1 + pkm2*xk
+		qk = qkm1 + qkm2*xk
+		pkm2 = pkm1
+		pkm1 = pk
+		qkm2 = qkm1
+		qkm1 = qk
+
+		if qk != 0 {
+			r = pk / qk
+		}
+		if r != 0 {
+			t = math.Abs((ans - r) / r)
+			ans = r
+		} else {
+			t = 1.0
+		}
+
+		if t < thresh {
+			return ans
+		}
+
+		k1 += 1.0
+		k2 -= 1.0
+		k3 += 2.0
+		k4 += 2.0
+		k5 += 1.0
+		k6 += 1.0
+		k7 += 2.0
+		k8 += 2.0
+
+		if (math.Abs(qk) + math.Abs(pk)) > big {
+			pkm2 *= biginv
+			pkm1 *= biginv
+			qkm2 *= biginv
+			qkm1 *= biginv
+		}
+		if (math.Abs(qk) < biginv) || (math.Abs(pk) < biginv) {
+			pkm2 *= big
+			pkm1 *= big
+			qkm2 *= big
+			qkm1 *= big
+		}
+	}
+	return ans
+}
+
+// Power series for incomplete beta integral. Use when b*x is small and x not
```

Comment:
`pseries returns the incomplete beta integral evaluated by a power series. Use ...`